### PR TITLE
Fix #1

### LINKED ISSUE #1: Include only-changes parameter
The aim of this parameter is only to show the report of the files changed in the PR.

We only need to add to the jestCommand the parameter:
```
--changeSince=context.payload.pull_request.base_ref
```

https://jestjs.io/docs/en/cli.html#--changedsince

### PR DESCRIPTION
**Pull request by osedoe**
🤞

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -60,3 +60,25 @@ export const createComment = async (comment: string) => {
 
 export const generateCoverageSummary = async (jestCommand: string): Promise<string> =>
   await execCommand(jestCommand, summaryFormatter);
+
+const generateChangeSinceParam = () => {
+  const param = getInput('only-changes');
+
+  switch (param) {
+    case 'true':
+      return '--changeSince=context.payload.pull_request.base_ref';
```
Comment (raulanatol) [inline src/utils.ts]: You need to check if is a pull_request action first.

```suggestion
      return `--changeSince=${context.payload.pull_request.base_ref}`;
```

File: src/utils.ts [hunk 2]
```diff
@@ -60,3 +60,25 @@ export const createComment = async (comment: string) => {
 
 export const generateCoverageSummary = async (jestCommand: string): Promise<string> =>
   await execCommand(jestCommand, summaryFormatter);
+
+const generateChangeSinceParam = () => {
+  const param = getInput('only-changes');
```
Comment (raulanatol) [inline src/utils.ts]: I think that can be better if we use here a wrapper as `getBooleanInput`

File: src/utils.ts [hunk 3]
```diff
@@ -60,3 +60,25 @@ export const createComment = async (comment: string) => {
...
+      return '';
+    default:
+      console.warn('only-changes parameter: ', param);
+      warning('You need to pass either "true" or "false" as only-changes parameter');
+  }
+};
+
+export const generateJestCommand = () => {
+  const baseCommand = getInput('jest-command');
+  const changeSinceParam = generateChangeSinceParam();
```
Comment (raulanatol) [inline src/utils.ts]: We should need to check if the 'changeSince' parameter already exists in the `jest-command` input first.

**Review by raulanatol — CHANGES_REQUESTED** [review]

**Comment by osedoe** [comment]
I'll get to it this evening. Thx!

File: src/utils.ts [hunk 4]
```diff
@@ -61,24 +61,37 @@ export const createComment = async (comment: string) => {
...
     default:
-      console.warn('only-changes parameter: ', param);
-      warning('You need to pass either "true" or "false" as only-changes parameter');
+      return undefined;
+  }
+};
+
+const generateChangeSinceParam = (baseCommand: string) => {
+  const param = getBooleanInput('only-changes');
+
```
Comment (raulanatol) [inline src/utils.ts:80]:
```suggestion
if (!param) {
   return '';
}
```

File: src/utils.ts [hunk 5]
```diff
@@ -61,24 +61,37 @@ export const createComment = async (comment: string) => {
...
+  if (param === undefined || !context.payload.pull_request?.base_ref) {
+    console.warn('only-changes parameter: ', param);
+    warning('You need to pass either "true" or "false" as only-changes parameter');
+    return '';
+  }
```
Comment (raulanatol) [inline src/utils.ts]:
```suggestion

```

File: src/utils.ts [hunk 6]
```diff
@@ -61,24 +61,37 @@ export const createComment = async (comment: string) => {
...
+
+  if (!param || baseCommand.includes('changeSince')) {
```
Comment (raulanatol) [inline src/utils.ts]:
```suggestion
  if (baseCommand.includes('changeSince')) {
```

File: src/utils.ts [hunk 7]
```diff
@@ -61,24 +61,37 @@ export const createComment = async (comment: string) => {
...
+    return '';
+  }
+
+  if (param && context.payload.pull_request?.base_ref) {
```
Comment (raulanatol) [inline src/utils.ts]:
```suggestion
  if (context.payload.pull_request?.base_ref) {
```

**Review by raulanatol — CHANGES_REQUESTED** [review]

**Review by raulanatol — APPROVED** [review]

**Pull request merged**